tests/phpunit/Civi/Financeextras/Payment/RefundTest.php

```diff
@@ -32,28 +31,33 @@ public function testIsEligibleForRefund() {
 
     $contribution = ContributionFabricator::fabricate($contributionParams);
 
-    $refundPayment = new Refund();
-    $isEligible = $refundPayment->isEligibleForRefund($contribution['id']);
+    $refundPayment = new Refund($contribution['id']);
+    $isEligible = $refundPayment->isEligibleForRefund();
     $this->assertTrue($isEligible);
 
   }
 
   public function testIsNotEligibleForRefund() {
     $contact = ContactFabricator::fabricate();
+    $paymentProcessor = PaymentProcessorFabricator::fabricate([
+      'payment_processor_type_id' => "FirstData",
```

Comment:
Do we really need this?  Contribution can be created without a payment processor attached to this.

Reply:
removed it also fixed code where its checking key, added isset check to remove / fix warning